tests/kernel/mem_protect/userspace/src/main.c

```diff
@@ -892,18 +905,19 @@ struct foo {
 
 struct foo stest_member_stack;
 
-void z_impl_stack_info_get(u32_t *start_addr, u32_t *size)
+void z_impl_stack_info_get(char **start_addr, size_t *size)
 {
-	*start_addr = k_current_get()->stack_info.start;
+	*start_addr = (char *)k_current_get()->stack_info.start;
```

Comment:
Shouldn't we be really treating `start_addr` here as `uintptr_t *`, as `_thread_stack_info` defines `start` as `uintptr_t`?

p.s. Speaking of which, it seems `_thread_stack_info` still defines `size` as `u32_t` rather than `size_t`.

Reply:
this is confusing as currently expressed -- I'll revise

Reply:
Actually we should not be using uintptr_t * at all.
The start member of thread_stack_info is stored as a uintptr_t because it is a memory address. It does not point to a uintptr_t.
I did fix the size member to use size_t.